geomstats/geometry/hypersphere.py

```diff
@@ -328,9 +384,14 @@ def extrinsic_to_intrinsic_coords(self, point_extrinsic):
         point_intrinsic : array-like, shape=[..., dim]
             Point on the hypersphere, in intrinsic coordinates.
         """
-        point_intrinsic = point_extrinsic[..., 1:]
+        if self.dim == 2:
+            return self.extrinsic_to_spherical(point_extrinsic)
+        if self.dim == 1:
+            return self.extrinsic_to_angle(point_extrinsic)
 
-        return point_intrinsic
+        raise NotImplementedError(
+            "Intrinsic coordinates are only implemented in dimension 1 and 2."
```

Comment:
we have spherical coordinates? could we make these intrinsic?

Reply:
Yes we do! What do you mean? I wrote that intrinsic corresponds to spherical if dim == 2